### ISSUE TITLE
FortiGate SSH Issue

### ISSUE THREAD

**Issue by ColvusBM** [warpdotdev/Warp#5030]
### Dupe Check

- [X] I have [searched Warp bugs](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3ABugs) and there are no duplicates

### Describe the bug

Unable to SSH onto FortiGate specifically with Warp. 

### To reproduce

SSH in every time, get disconnected. 

### Screenshots

<img width="1136" alt="Screenshot 2024-06-15 at 20 40 43" src="https://github.com/warpdotdev/Warp/assets/55392293/2cb52a0f-449a-48cd-8fad-38e05b64bfe5">

### Operating system

MacOS

### Operating system and version

OSX

### Regression

No, this bug or issue has existed throughout my experience using Warp

### Does this block you from using Warp daily?

Yes, this issue prevents me from using Warp daily.

### Is this a Warp specific issue? (i.e. does it happen in Terminal, iTerm, Kitty, etc.)

Yes, this I confirmed this only happens in Warp, not other terminals.

### Warp Internal (ignore): linear-label:b9d78064-c89e-4973-b153-5178a31ee54e

None

**Comment by dannyneira** [comment]
Hi @ColvusBM sorry to hear about this. Warp has an SSH Wrapper that enables features (like the blocks, prompt, completions, etc.) on remote machines requiring the default prompt on the remote machine to be either `bash` or `zsh`.

As a workaround you can add `command ssh` to your `Settings > Subshells > Added commands` menu, then run `command ssh <user@server>` to connect to remote session, this will attempt to enable Warp features as a subshell.

You can also try disabling the SSH Wrapper by navigating to `Settings > Features`. Blocks will stop working but with it disabling, SSH should work as normal.

Related docs:
[https://docs.warp.dev/features/ssh](https://docs.warp.dev/features/ssh#ssh-wrapper-fails)
https://docs.warp.dev/features/subshells

If those don't work, please let us know.

**Comment by ColvusBM** [comment]
Good afternoon, The first option did the very same thing.  The second option worked. Will this be fixed? RegardsBrandon. Sent from my iPhoneOn 22 Jun 2024, at 00:40, Danny Neira ***@***.***> wrote:﻿
Hi @ColvusBM sorry to hear about this. Warp has an SSH Wrapper that enables features (like the blocks, prompt, completions, etc.) on remote machines requiring the default prompt on the remote machine to be either bash or zsh.
As a workaround you can add command ssh to your Settings > Subshells > Added commands menu, then run command ssh ***@***.***> to connect to remote session, this will attempt to enable Warp features as a subshell.
You can also try disabling the SSH Wrapper by navigating to Settings > Features. Blocks will stop working but with it disabling, SSH should work as normal.
Related docs:
https://docs.warp.dev/features/ssh
https://docs.warp.dev/features/subshells
If those don't work, please let us know.

**Comment by dannyneira** [comment]
Yes we plan to improve the SSh experience in an up coming cycle. We appreciate your patience as we work to make Warp better.

**Comment by peterge-misoft** [comment]
For me, the problem does show up on FG101F with latest FortiOS (v7.6.3) and Warp v0.2025.07.09.08.11.stable_01:

```
ssh fgcgn.misoft.lan
fortigatecologne # fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # 
command parse error before '$SHELL'
Command fail. Return code -61

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # > > > > > > > > > > > 
command parse error before '-a'
Command fail. Return code -61

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Unknown action 0

fortigatecologne # Connection to fgcgn.misoft.lan closed.
```
